### PR TITLE
Fix reloading page issue when user make Stake S3F/ Unstake S3F actions on page “Earn SNOB” 

### PR DESCRIPTION
**Pull request by thongpt-codelink**
## Status
**Done**

## Description
- The logic for fetching information and rendering the poolS3F is now separated, no need to wait until other fetching resolved.
- Instead of reloading the page when the action is submitted, fetching only relevant information and re-rendering the element that the action interact.

## Reference
Demo site: https://vfat-tools-rho.vercel.app/earn/

https://user-images.githubusercontent.com/83624863/117408623-6442de80-af3a-11eb-8ef0-921269c9bb32.mov

### CODE REVIEW

File: src/static/js/earn.js [hunk 1]
```diff
@@ -73,6 +74,113 @@ async function main() {
   const LINK_AVAX_TVL = "https://info.pangolin.exchange/#/account/0x974Ef0bDA58C81F3094e124f530eF34fe70dc103"
   const USDT_AVAX_TVL = "https://info.pangolin.exchange/#/account/0x74dB28797957a52a28963F424dAF2B10226ba04C"
 
+  const stakingContract_stake = async ({
```
Comment (thongpt-codelink) [inline src/static/js/earn.js:77]: This function is on line **2315** in the original.

File: src/static/js/earn.js [hunk 2]
```diff
@@ -73,6 +74,113 @@ async function main() {
...
+          `ok`,
+          false
+        );
+      }
+    } else {
+      snobMessage(`Oops! Failed`, `You have no tokens to stake`, `close-circle-outline`, `danger`, false, `ok`, false);
+    }
+  }
+
+  const stakingContract_withdraw = async ({STAKING_ABI, STAKING_ADDR, App, SNOB_TOKEN, S3F_TOKEN, renderPoolS3F}) => {
```
Comment (thongpt-codelink) [inline src/static/js/earn.js:135]: This function is on line **2354** in the original.

**Review by huynvk — APPROVED** [review]
LGTM

**Comment by best-coder-NA** [comment]
Thanks everyone! Appreciate the updates.

**Pull request merged**